### PR TITLE
Add average encoder

### PR DESCRIPTION
**Pull request by himkt**
I implemented the encoder that averages word vectors.

### CODE REVIEW

**Comment by himkt** [comment]
CI passed!
@jun-harashima Please check it, please. :bow:

**Comment by himkt** [comment]
- Support AverageEncoder
- Add the abstract Encoder

**Comment by himkt** [comment]
[NOTE] Currently, test specs pass **only** on CPU.

File: msnc/encoder.py [hunk 1]
```diff
@@ -82,3 +106,43 @@ def _view(self, H):
...
+        H = []
+        for x in X:
+            h = self._embed(self.util.tensorize(x))
+            h = h.mean(dim=0)
+            H.append(h)
+
+        return torch.stack(H, dim=0)
+
+    def _embed(self, X):
+        return self.embedding(X)
```
Comment (jun-harashima) [inline msnc/encoder.py]: We can merge the method with `_embed` of `RecurrentEncoder`.

**Comment by jun-harashima** [comment]
LGTM with a comment 🙏

File: msnc/encoder.py (hunk 1 above)
Comment (himkt) [reply msnc/encoder.py]: Ah, you're right, thanks! :bow:

**Comment by himkt** [comment]
Thanks for the comment!
I updated my PR and merge it now.

**Pull request merged**